.github/workflows/ci.yml

```diff
@@ -5,143 +5,314 @@ concurrency:
+      - name: Ensure line endings are consistent
+        run: git config --global core.autocrlf input
+      - name: Use Node.js ${{ env.DEFAULT_NODE_VERSION }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.DEFAULT_NODE_VERSION }}
+      - name: Enable Corepack
+        run: corepack enable
+      - name: Ensure line endings are consistent
+        run: git config --global core.autocrlf input
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Load Yarn cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            .yarn
+            **/node_modules
+          key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+      - name: Install project dependencies
+        run: yarn install --immutable
+      - name: Run linter
+        run: yarn run lint
+
   test:
-    runs-on: ${{ matrix.os }}
+    name: Tests
+    needs: lint
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
+        os:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
         node-version:
           - 18.x
           - 20.x
           - 22.x
+    runs-on: ${{ matrix.os }}
     steps:
+      - name: Ensure line endings are consistent
+        run: git config --global core.autocrlf input
       - name: Use Node.js ${{ matrix.node-version }}
         uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node-version }}
-      - name: Ensure line endings are consistent
-        run: git config --global core.autocrlf input
+      - name: Enable Corepack
+        run: corepack enable
       - name: Check out repository
         uses: actions/checkout@v4
-      - name: Load cache
+      - name: Load Yarn cache
         uses: actions/cache@v4
         with:
           path: |
+            .yarn
             **/node_modules
-            .rdf-test-suite-cache
-            .rdf-test-suite-ldf-cache
-          key: ${{ runner.os }}-test-modules-${{ secrets.CACHE_VERSION }}-${{ hashFiles('**/yarn.lock') }}
-      - name: Setup firefox
-        uses: browser-actions/setup-firefox@latest
-        with:
-          firefox-version: latest
-      - name: Install dependencies
-        run: yarn install --frozen-lockfile
-      - name: Build project
-        run: yarn run build
-      - name: Build project -- esModuleInterop true
-        run: yarn run build:ts --esModuleInterop true
-      - name: Run depcheck
-        run: yarn run depcheck
-      - name: Run tests
-        run: yarn run test:ci
-      - name: Run browser tests
-        if: ${{ matrix.os == 'ubuntu-latest' }}
-        uses: nick-fields/retry@v3
-        with:
-          timeout_minutes: 2
-          max_attempts: 3
-          command: yarn run test:browser
+          key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+      - name: Install project dependencies
+        run: yarn install --immutable
+      - name: Compile TypeScript
```

Comment:
This and the following steps are not necessary, as they should be invoked automatically upon calling `yarn install`.

Reply:
Also applies to all other jobs in this workflow.

Reply:
I actually adjusted the package `postinstall` script to not build everything by default, because it was doing it in the CI, as well, which is unnecessary for the lint task, hence the splitting up also here. Now it explicitly builds when the user instructs it to build. I thought it would be clearer this way, and would help better illustrate the need to build after changes. :slightly_smiling_face:

Reply:
IMO, the dev environment should be fully ready after running `install`.
Requiring users to still execute a `build` script is counter-intuitive IMO, so I would prefer if we keep this.